### ISSUE TITLE
Page list completeness

### ISSUE THREAD

**Issue by mattgarrish** [w3c/epub-specs#1503]
As originally asked in #1471:

> do content creators have to list all pages, or can they list only some of them (e.g. the page number at the beginning of the chapter)

**Comment by mattgarrish** [comment]
This issue may be handled by the upcoming wcag success criteria for a page list. It requires that a means of accessing all the included page breaks be provided. So whatever we require/recommend/advise for including page breaks will have to be reflected in the page list to reach AA.

What gives me a bit of pause is why we require a page list but have page breaks as optional. Maybe we should reconsider what our success criteria says in light of what wcag will soon require?

In other words, maybe we should drop the requirement for a page list when there aren't page markers?

Most epubs I've encountered have the markers if they have a page list, so when is there a page list without markers? That seems like it would be harder to produce. Or is this requirement directed at full audio epubs with minimal text, although I'm not even sure how that would work? Could that be handled in epub audio guidelines instead if so?

/cc @avneeshsingh @GeorgeKerscher @gregoriopellegrino @clapierre

**Comment by gregoriopellegrino** [comment]
For the remediation of some EPUBs I've made a page list without page markers (it was too time expensive), giving the user the number of the first page of each chapter.

If we move forward with issue #1502 then this technique will no longer be possible and having a page list will be equivalent to having page breaks. 

Actually there is an edge case: in FXL EPUBs there might be a full page list, without any page breaks.

**Comment by mattgarrish** [comment]
> For the remediation of some EPUBs I've made a page list without page markers (it was too time expensive)

Right, this is kind of what I'm questioning. Is it necessary to force a page list for content that doesn't have breaks if all the breaks aren't going to be accounted for? From your description it's doubtful the page list would be terribly useful since it's effectively the same as the table of contents but much less clear.

> Actually there is an edge case: in FXL EPUBs there might be a full page list, without any page breaks.

Good point. This is the outlier.

**Comment by mattgarrish** [comment]
Maybe there's a way we can reformulate the requirements more along WCAG lines so it's clearer what you have to meet for a given level.

For example:

- Level A - you must include a page list but we don't impose any further requirements on it - this would match our current requirement
- Level AA - the page list must account for every page of content reproduced from the source - this wouldn't require blank pages or omitted content be listed
- Level AAA - the page list must account for every page from the source including any blank pages and omitted content

Would this make sense or are we getting too strict?

**Comment by gregoriopellegrino** [comment]
Wow, great for me!

**Comment by clapierre** [comment]
I like the idea, but my only concern we are now tying EPUB Accessibility requirements to WCAG levels.  It kinda feels like there is an EPUB - Level A/AA/AAA, but I think that would be too complicated.

Therefore if an EPUB really meets all of WCAG 2.1 AA requirements, but the page list is say missing some content, then the EPUB would be downgraded to WCAG 2.1 A.  Likewise way the EPUB meets WCAG 2.1 A but they accurately have a page list with all pages including blank pages I don't think we could elevate them to WCAG 2.1 AAA status.  So I am not sure how this could work.

**Comment by mattgarrish** [comment]
> It kinda feels like there is an EPUB - Level A/AA/AAA, but I think that would be too complicated.

It's not making official tiers for EPUB, but raising the bar of EPUB conformance the higher you want to claim WCAG conformance.

> Therefore if an EPUB really meets all of WCAG 2.1 AA requirements, but the page list is say missing some content, then the EPUB would be downgraded to WCAG 2.1 A.

To make a claim you're fully conforming with the EPUB specification, yes. It doesn't change the actual WCAG conformance level.

But this problem already exists in our having separate requirements. You can meet WCAG AA and completely fail the EPUB specification for missing any requirement we have. Likewise, you can meet only Level A but then will have to implement whatever extra requirements we define.

You can also claim conformance without having to implement the additional requirements of the EPUB spec, or only doing some bare minimum to pay lip service to them, because we can't layer the requirements the same way WCAG does.

> I don't think we could elevate them to WCAG 2.1 AAA

Again, we're not actually changing the WCAG conformance. To meet AAA, the publisher would have to first meet all of WCAG's requirements.

AAA is more often aspirational, as the goal is to get people to implement these success criteria when they can. That's why the AAA equivalents for the page list and markers would be the strictest. We'd like to see this done, but it's an acknowledgement it's not going to be feasible for everyone to do.

**Comment by avneeshsingh** [comment]
If we get into three levels for static pages, then we would end up in multiple levels of requirements for other EPUB accessibility features also. This will make it quiet confusing. We already have six variables, WCAG 2.0, WCAG 2.1, WCAG a, AA and AAA. If we end up with 3 more levels for EPUB accessibility, then it will take the combinations to 18!

It would be better to differentiate requirements through must / should.
- One must include a page list if page markers are used but we don't impose any further requirements on it. (this is slated for WCAG 2.2)
- One should include page markers at positions equivalent to the print equivalent or fixed layout equivalent. (it is well explained in current EPUB accessibility specs)
And we can emphasize the need of having all pages in the techniques (which is level AAA mentioned by Matt)

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/publishing/groups/epub-wg/Meetings/Minutes/2021-03-11-epub-a11y#section4) on 2021-03-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

### 4. pages related issues 

_See github issue [#1503](https://github.com/w3c/epub-specs/issues/1503), [#1502](https://github.com/w3c/epub-specs/issues/1502), [#1501](https://github.com/w3c/epub-specs/issues/1501), [#1500](https://github.com/w3c/epub-specs/issues/1500), [#1542](https://github.com/w3c/epub-specs/issues/1542)._



**Matt Garrish:**: Updated EPUB accessibility for addressing a part of the following static pages issues. Some requirements can be addressed by EPUB Accessibility while others should go to best practices.

> *Avneesh Singh:* [https://w3c.github.io/epub-specs/epub33/a11y/index.html#sec-page-nav](https://w3c.github.io/epub-specs/epub33/a11y/index.html#sec-page-nav)

**Matt Garrish:** we had different questions about pagelists  
… one propose is to structure the requirements for pagelists  
… e.g. level A: some pages in the pagelist  
… level AA: all content pages linked in pagelist  
… level AAA: all pages (no exclusion, even blank pages)  
… I didn't move over with my requirements  

**Avneesh Singh:** I don't think these issues can be added in the guidelines, maybe the best place is best practices  

**Tzviya Siegman:** we had similar in PDFs with "this page intentionally left blank", where we left blank pages for starting the chapters on page right  
… I think that marking blank pages can be really confusing  
… maybe we can use ARIA  

**Avneesh Singh:** yes, but then someone will say that ARIA is not only for AT  

**Charles LaPierre:** I'm not sure if Read Aloud by the Reading system can get that information  

**Matt Garrish:** for sure we can investigate the use of ARIA label or something similar  

**Avneesh Singh:** I think this task force should focus on the requirements Matt put in the issues  
… we can go through the issue tracker  
… and discuss there  

**Matt Garrish:** we have a issue about the ordering of the pagelist  
… if I have the pages moved in the digital version, what happen to the pagelist?  

**Tzviya Siegman:** I have several examples of this  

**Avneesh Singh:** I think for AT users it would be useful to have an alert  

**Charles LaPierre:** maybe we can put in in the accessibility summary (metadata), but do we need to have some requirements for reading systems?  

**Gregorio Pellegrino:** maybe something like play order attribute in NCX tocs  

---


</details>

**Comment by mattgarrish** [comment]
As discussed, closing this with the changes in https://github.com/w3c/epub-specs/pull/1558
